model/labelset_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2013 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func TestUnmarshalJSONLabelSet(t *testing.T) {
+	type testConfig struct {
+		LabelSet LabelSet `yaml:"labelSet,omitempty"`
+	}
+
+	// valid LabelSet JSON
+	labelSetJSON := `{
+	"labelSet": {
+		"monitor": "codelab",
+		"foo": "bar"
+	}
+}`
+	var c testConfig
+	err := json.Unmarshal([]byte(labelSetJSON), &c)
```

Comment:
Hmm, it this code dead yet?

Reply:
I am not sure I follow :).

Reply:
I'm wondering if we still have code that uses this, and if not if we can delete it.

Reply:
I'll update the year in this file's license header

Reply:
Are you referring to the JSON Unmarshal specifically ? LabelSet is being used widely.